src/legacy/server/config/schema.js

```diff
@@ -100,6 +100,8 @@ export default () => Joi.object({
       otherwise: Joi.default(false),
     }),
 
+    ignoreElasticsearchVersionMismatch: Joi.boolean().default(false),
```

Comment:
I'm not attached to this name, so if someone can think of something better. Other suggestions: 
 - `allowIncompatibleClusterNodes`

Reply:
Mind if we move this to the Elasticsearch plugin, which defines the [healthcheck](https://github.com/elastic/kibana/blob/master/src/core/server/elasticsearch/elasticsearch_config.ts#L67)?

Something like `elasticsearch.healthCheck.ignoreVersionMismatch`

Reply:
Sure, I'll move it 👍

Reply:
Done.